src/interfaces/CachingMechanism.ts

```diff
@@ -9,7 +11,11 @@ export interface CachingMechanismInterface {
    * @param key The key to retrieve.
    * @returns The value if it exists, otherwise null.
    */
-  get<T>(key: string): Promise<T | null>;
+  get<ObjectType, OverrideType>(
```

Comment:
I'm kind of confused how these generics work. For example with the RedisClient, every time I want to `get` a value do I have to use the syntax `get<ModifiedFlow[]>`?

I'm also confused how this OverrideType works